src/components/scenes/EdgeLoginScene.tsx

```diff
@@ -1,242 +1,158 @@
-import { EdgeLobby } from 'edge-core-js'
 import * as React from 'react'
-import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
-import FastImage from 'react-native-fast-image'
-import { isIPhoneX } from 'react-native-safe-area-view'
+import { ActivityIndicator, Image, View } from 'react-native'
 import { sprintf } from 'sprintf-js'
 
 import { lobbyLogin } from '../../actions/EdgeLoginActions'
+import { useHandler } from '../../hooks/useHandler'
 import s from '../../locales/strings'
-import { PrimaryButton } from '../../modules/UI/components/Buttons/PrimaryButton.ui'
-import { SecondaryButton } from '../../modules/UI/components/Buttons/SecondaryButton.ui'
 import { config } from '../../theme/appConfig'
-import { THEME } from '../../theme/variables/airbitz'
-import { connect } from '../../types/reactRedux'
+import { useDispatch, useSelector } from '../../types/reactRedux'
 import { NavigationProp } from '../../types/routerTypes'
 import { SceneWrapper } from '../common/SceneWrapper'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { TitleText } from '../text/TitleText'
+import { Alert } from '../themed/Alert'
+import { MainButton } from '../themed/MainButton'
 
-type OwnProps = {
+type Props = {
   navigation: NavigationProp<'edgeLogin'>
 }
-type StateProps = {
-  error: string | null
-  isProcessing: boolean
-  lobby: EdgeLobby | null
-}
 
-type DispatchProps = {
-  accept: () => void
-}
+export const EdgeLoginScene = (props: Props) => {
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
 
-type Props = StateProps & DispatchProps & OwnProps
+  const error = useSelector(state => state.core.edgeLogin.error)
+  const isProcessing = useSelector(state => state.core.edgeLogin.isProcessing)
+  const lobby = useSelector(state => state.core.edgeLogin.lobby)
+  const dispatch = useDispatch()
```

Comment:
I would love to pull all the state out of Redux - pass the login token as part of the route props, and then use `React.useState` to manage all this stuff, instead of Redux and actions.